tests/testthat/test_get_sim.R

```diff
@@ -0,0 +1,36 @@
+test_that("test get_sim functionalities", {
+
+  library(serofoi)
+
+  tsur <- 2050
+  birth_year_min <- 2000
+  survey_label <- "foi_sim"
+  country = 'None'
+  test = "test"
+  antibody = "IgG"
+
+  sim_data <- data.frame(birth_year = c(birth_year_min:(tsur - 1))) %>%
+    mutate(tsur = tsur,
+           country = country,
+           test = test,
+           antibody = antibody,
+           survey = survey_label,
+           age_mean_f = tsur - birth_year)
+
+  #----- Test function generate_sim_probability
+  foi_sim <- rep(0.02, 50)
+  sim_probability <- get_sim_probability(sim_data = sim_data,
+                                         foi = foi_sim)
+  expect_s3_class(sim_probability, "data.frame")
+  expect_type(sim_probability$age, "integer")
+  expect_type(sim_probability$probability, "double")
```

Comment:
We should also be testing the values (i.e. the probabilities) that are produced by this function.

Reply:
What I did here is to generate the exposure matrix and the probabilities by hard for the simple case of constant FoI and compare them with the output of the function:
```
n_years <- 50
foi_sim <- rep(0.02, n_years)
sim_probability <- get_sim_probability(sim_data = sim_data,
                                       foi = foi_sim)

exposure_matrix <- matrix(1, n_years, n_years)
exposure_matrix[lower.tri(exposure_matrix)] <- 0
probabilities <- purrr::map_dbl(1:n_years, ~1-exp(-pracma::dot(exposure_matrix[., ], foi_sim)))

expect_true(all(probabilities == sim_probability$probability))
```
Is this in accordance to what you had in mind?

Reply:
Yep, exactly.